core/middlewares/with-analytics-cookies.ts

```diff
@@ -0,0 +1,18 @@
+import { setVisitIdCookie, setVisitorIdCookie } from '~/lib/analytics/bigcommerce';
+
+import { MiddlewareFactory } from './compose-middlewares';
+
+export const withAnalyticsCookies: MiddlewareFactory = (next) => {
+  return async (request, event) => {
+    await setVisitorIdCookie();
+    await setVisitIdCookie(onNewVisit);
+
```

Comment:
🍹 Instead of a callback, can we return a new visit id value and handle that within the middleware request? It'll be easier to use `event.waitUntil` within the middleware body than having to pass it as an argument.

For example, you can trigger an API request without blocking the initial render using `event.waitUntil`:
```ts
async (request, event) => {
  await setVisitorIdCookie();
  
  const newVisitId = await setVisitIdCookie();

  if (newVisitId) {
    // middleware will continue and won't end the process until after the event is triggered
    event.waitUntil(() => {
      await sendShopperVisitEvent(newVisitId);
    });
  }
}
```